### PR TITLE
[MLU] add mlu device for infer

### PR DESCRIPTION
**Pull request by PeiyuLau**
add mlu device for infer

### CODE REVIEW

File: tools/infer/utility.py
Comment (onecatcn) [inline tools/infer/utility.py:1]: There is some discrepency between the codes for npu and mlu, please justify the difference.

File: tools/infer/utility.py [hunk 1]
```diff
@@ -245,8 +249,9 @@ def create_predictor(args, mode, logger):
                     logger.info(E)
                     logger.info("Please keep your paddlepaddle-gpu >= 2.3.0!")
 
-        elif args.use_npu:
-            config.enable_custom_device("npu")
+        elif args.use_custom_device:
+            dev_type = paddle.device.get_all_custom_device_type()[0]
+            config.enable_custom_device(dev_type)
```
Comment (shiyutang) [inline tools/infer/utility.py:254]: The custom device should not be able to replace npu.

It seems you should add mlu device on top of the original code with another elif clause.

**Review by shiyutang — CHANGES_REQUESTED** [review]

File: tools/infer/utility.py [hunk 2]
```diff
@@ -37,6 +37,11 @@ def init_args():
     parser.add_argument("--use_gpu", type=str2bool, default=True)
     parser.add_argument("--use_xpu", type=str2bool, default=False)
     parser.add_argument("--use_npu", type=str2bool, default=False)
+    parser.add_argument(
+        "--use_custom_device",
```
Comment (shiyutang) [inline tools/infer/utility.py:41]: If you want to add mlu, I suppose it is better if this should is use_mlu.

File: tools/infer/utility.py (hunk 1 above)
Comment (shiyutang) [reply tools/infer/utility.py:257]: config.enable_custom_device("mlu")

**Review by shiyutang — CHANGES_REQUESTED** [review]
any custom device is vague and should be replaced with mlu

**Review by shiyutang — APPROVED** [review]
LGTM

**Pull request merged**